tests/integration/update_cluster/launch_templates/kubernetes.tf

```diff
@@ -0,0 +1,733 @@
+}
+
+resource "aws_autoscaling_group" "master-us-test-1b-masters-launchtemplates-example-com" {
+  name                 = "master-us-test-1b.masters.launchtemplates.example.com"
+  launch_configuration = "${aws_launch_configuration.master-us-test-1b-masters-launchtemplates-example-com.id}"
+  max_size             = 1
+  min_size             = 1
+  vpc_zone_identifier  = ["${aws_subnet.us-test-1b-launchtemplates-example-com.id}"]
+
+  tag = {
+    key                 = "KubernetesCluster"
+    value               = "launchtemplates.example.com"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "Name"
+    value               = "master-us-test-1b.masters.launchtemplates.example.com"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "k8s.io/role/master"
+    value               = "1"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "kops.k8s.io/instancegroup"
+    value               = "master-us-test-1b"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "kubernetes.io/cluster/launchtemplates.example.com"
+    value               = "owned"
+    propagate_at_launch = true
+  }
+
+  metrics_granularity = "1Minute"
+  enabled_metrics     = ["GroupDesiredCapacity", "GroupInServiceInstances", "GroupMaxSize", "GroupMinSize", "GroupPendingInstances", "GroupStandbyInstances", "GroupTerminatingInstances", "GroupTotalInstances"]
+}
+
+resource "aws_autoscaling_group" "master-us-test-1c-masters-launchtemplates-example-com" {
+  name                 = "master-us-test-1c.masters.launchtemplates.example.com"
+  launch_configuration = "${aws_launch_configuration.master-us-test-1c-masters-launchtemplates-example-com.id}"
+  max_size             = 1
+  min_size             = 1
+  vpc_zone_identifier  = ["${aws_subnet.us-test-1c-launchtemplates-example-com.id}"]
+
+  tag = {
+    key                 = "KubernetesCluster"
+    value               = "launchtemplates.example.com"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "Name"
+    value               = "master-us-test-1c.masters.launchtemplates.example.com"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "k8s.io/role/master"
+    value               = "1"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "kops.k8s.io/instancegroup"
+    value               = "master-us-test-1c"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "kubernetes.io/cluster/launchtemplates.example.com"
+    value               = "owned"
+    propagate_at_launch = true
+  }
+
+  metrics_granularity = "1Minute"
+  enabled_metrics     = ["GroupDesiredCapacity", "GroupInServiceInstances", "GroupMaxSize", "GroupMinSize", "GroupPendingInstances", "GroupStandbyInstances", "GroupTerminatingInstances", "GroupTotalInstances"]
+}
+
+resource "aws_autoscaling_group" "nodes-launchtemplates-example-com" {
```

Comment:
This ASG resource doesnt have the `launch_template` property that associates it with the `aws_launch_template` defined below.

https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html#with-latest-version-of-launch-template

I know your other PR mentions that LT werent working at all, but it seems it only fixes CF, not TF. It may be out of scope for this PR but we should definitely fix that because it seems that launch templates wont work with terraform at all currently.

Side note: i'm hoping that https://github.com/kubernetes/kops/issues/8648 will prevent this from happening again in the future, since I believe this would be caught during a `terraform validate` or `terraform plan`

Reply:
Yes, that was the main problem with TF. The other PR should fix it with this:
https://github.com/kubernetes/kops/blob/1294d975d478ed7ba18d9fe9d53110ab11578ddb/upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go#L736-L740